requests_toolbelt/adapters/source.py

```diff
@@ -60,3 +60,7 @@ def init_poolmanager(self, connections, maxsize, block=False):
             maxsize=maxsize,
             block=block,
             source_address=self.source_address)
+
+    def proxy_manager_for(self, *args, **kwargs):
+        kwargs['source_address'] = self.source_address
```

Comment:
What if instead we do

```py
kwargs.setdefault('source_address', self.source_address)
```

I suppose it's possible that's already being sent and we wouldn't want to forcibly override it? I'm not sure.